examples/vision/attention_mil_classification.py

```diff
@@ -0,0 +1,589 @@
+- Some of attention operator code implementation was inspired from https://github.com/utayao/Atten_Deep_MIL.
+- Imbalanced data [tutorial](https://www.tensorflow.org/tutorials/structured_data/imbalanced_data)
+by TensorFlow.
+
+## What will you learn in this example?
+The approach to find a target feature (without explicitly labelling them in prior)
+among a bag of features will be taught. You will mainly learn:
+
+* about attention-based deep MIL, its applications (applied on MNIST dataset in the
+example; can be applied on other image classification datasets) and its results'
+interpretability.
+* that its a weakly supervised learning algorithm which means it will be of very
+helpful if you suffer from weakly labelled data.
+"""
+"""
+## Setup
+"""
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+from tqdm import tqdm
+from matplotlib import pyplot as plt
+
+plt.style.use("ggplot")
+
+"""
+## Create dataset
+At first we will set up the configurations and then prepare the datasets.
+
+We will create a set of bags and assign their labels accordingly. This is performed by choosing
+a positive instance and forming associated bags of instances. If atleast one positive instance
+is available in a bag, the bag is considered as a positive bag. If it does not contain any
+positive instance, the bag will be considered as negative.
+
+### Configurations
+
+- `POSITIVE_CLASS`: The desired class to be kept in the positive bag.
+- `BAG_COUNT`: The number of training bags.
+- `VAL_BAG_COUNT`: The number of validation bags.
+- `BAG_SIZE`: The number of instances in a bag.
+- `PLOT_SIZE`: The number of bags to plot.
+- `ENSEMBLE_AVG_COUNT`: The number of models to create and get their average. (OPTIONAL:
+often results in better performance - set to 1 for single model)
+"""
+
+POSITIVE_CLASS = 1
+BAG_COUNT = 1000
+VAL_BAG_COUNT = 300
+BAG_SIZE = 3
+PLOT_SIZE = 3
+ENSEMBLE_AVG_COUNT = 5
+
+"""
+### Prepare bags
+Since the attention operator is a permutation-invariant operator, an instance with a
+positive class label is randomly placed among the instances in the positive bag.
+"""
+
+
+def create_bags(input_data, input_labels, positive_class, bag_count, instance_count):
+
+    # Set up bags.
+    bags = []
+    bag_labels = []
+
+    # Normalize input data.
+    input_data = np.divide(input_data, 255.0)
+
+    # Count positive samples.
+    count = 0
+
+    # Take out the filter for specific class.
+    filter_class = np.where(input_labels == positive_class)[0]
+
+    # Assign new variables consisting of this class.
+    data_positive_class = input_data[filter_class]
+    labels_positive_class = input_labels[filter_class]
+
+    # From overall data, remove this class.
+    data_negative_classes = np.delete(input_data, filter_class, 0)
+    labels_negative_classes = np.delete(input_labels, filter_class, 0)
+
+    # Merge both inputs and labels to each another.
```

Comment:
Not necessarily positive class comes first; since I am shuffling them in the next lines. I am just combining the data and labels of different classes together such that `data` consists of both classes and so for `labels`. I tried clarifying it in my recent commit.